### PR TITLE
frontend/fix: #446 resolved bugs for customer and driver

### PR DESCRIPTION
**Pull request by ashkriti**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Resolved the bugs :
1. In driver and customer Empty Rides error msg should display in the chosen language (BUG_48)
2. When customer is on ride then book ride in my rides empty section should be removed (BUG_54)
3. User should not get Keypad after giving the grant access while login into the app (BUG_64)
4. When Driver trying to login for first time after app is installed, on permission screen driver should not get keypad (BUG_66)

## Motivation and Context
https://github.com/nammayatri/nammayatri/issues/446

## How did you test it?
Tested manually

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Frontend/ui-customer/src/Flow.purs [hunk 1]
```diff
@@ -392,6 +392,7 @@ accountSetUpScreenFlow = do
 homeScreenFlow :: FlowBT String Unit
 homeScreenFlow = do
   (GlobalState currentState) <- getState
+  _ <- pure $ hideKeyboardOnNavigation true
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Flow.purs:395]: did you check this change in search location model if you do like this keyboard will close for every input

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]

**Comment by Vignesh-772** [comment]
get it tested

**Review by witcher-shailesh — CHANGES_REQUESTED** [review]

File: Frontend/ui-customer/src/Screens/CustomerUtils/MyRidesScreen/ComponentConfig.purs [hunk 2]
```diff
@@ -91,6 +92,7 @@ errorModalConfig = let
       , color = Color.yellow900
       , fontStyle = FontStyle.medium LanguageStyle
       , textSize = FontSize.a_16
+      , visibility = if (isLocalStageOn ST.HomeScreen) then VISIBLE else GONE
```
[the remark this answers is not in the document]
Comment (ashkriti) [reply Frontend/ui-customer/src/Screens/CustomerUtils/MyRidesScreen/ComponentConfig.purs:95]: Earlier the book now button was visible in the MyRidesScreen when the ride was ongoing for the first time and was not completed or cancelled. So when the local stage is home screen then only it should be visible.

File: Frontend/ui-customer/src/Screens/CustomerUtils/MyRidesScreen/ComponentConfig.purs [hunk 3]
```diff
@@ -25,6 +25,7 @@ import Language.Types (STR(..))
 import PrestoDOM (Length(..), Margin(..), Padding(..), Visibility(..))
 import Screens.Types as ST
 import Styles.Colors as Color
+import Storage (isLocalStageOn)
```
[the remark this answers is not in the document]
Comment (ashkriti) [reply Frontend/ui-customer/src/Screens/CustomerUtils/MyRidesScreen/ComponentConfig.purs:28]: Sure

**Review by Vignesh-772 — APPROVED** [review]

**Review by witcher-shailesh — APPROVED** [review]

**Pull request merged**